tests/expressions/tests.py

```diff
@@ -2018,6 +2018,13 @@ def test_non_empty_group_by(self):
         self.assertEqual(group_by_cols, [expr.expression])
         self.assertEqual(group_by_cols[0].output_field, expr.output_field)
 
+    def test_decimal_filter(self):
+        output_field = DecimalField(max_digits=20, decimal_places=17)
+        Number.objects.create(integer=0)
+        Number.objects.annotate(
+            x=ExpressionWrapper(Value(1.1), output_field=output_field),
+        ).filter(x=1.1).filter(x=Decimal('1.1')).get()
```

Comment:
Good point, comparing floats (or float-derived values) for equality is not a good idea. I've replaced all numbers with integers to prevent this, now things work as expected. I also made some other minor changes, now all tests pass as expected.